src/main/scala/org/hammerlab/guacamole/commands/jointcaller/VCFOutput.scala

```diff
@@ -114,100 +137,244 @@ object VCFOutput {
-      (if (includePooledNormal)
-        Seq(Input(samplesEvidence.normalDNAPooledIndex, "pooled_normal", "[no path]", TissueType.Normal, Analyte.DNA))
-      else Seq.empty) ++
-      (if (includePooledTumor)
-        Seq(Input(samplesEvidence.tumorDNAPooledIndex, "pooled_tumor", "[no path]", TissueType.Tumor, Analyte.DNA))
-      else Seq.empty)
+    val effectiveInputs =
+      subInputs.items ++
+      (
+        if (includePooledNormal)
+          Seq(
+            Input(
+              samplesEvidence.normalDNAPooledIndex,
+              "pooled_normal",
+              "[no path]",
+              TissueType.Normal,
+              Analyte.DNA
+            )
+          )
+        else
+          Seq.empty
+      ) ++
+      (
+        if (includePooledTumor)
+          Seq(
+            Input(
+              samplesEvidence.tumorDNAPooledIndex,
+              "pooled_tumor",
+              "[no path]",
+              TissueType.Tumor,
+              Analyte.DNA
+            )
+          )
+        else
+          Seq.empty
+      )
 
     val genotypes = effectiveInputs.map(input => {
+
       def alleleToString(x: String) = if (x.isEmpty) "." else x
-      def mixtureToString(mixture: Map[String, Double]) = mixture.map(pair => {
-        "%s->%.2f".format(alleleToString(pair._1), pair._2)
-      }).mkString("|")
+
+      def mixtureToString(mixture: AlleleMixture) =
+        (for {
+          (allele, vaf) <- mixture
+        } yield
+          "%s->%.2f".format(alleleToString(allele), vaf)
+        ).mkString("|")
+
+      def mixturesToString(posteriors: Map[AlleleMixture, Double]): String =
+        (for {
+          (mixture, probability) <- posteriors
+        } yield
+          "[%s]=%.8g".format(
+            mixtureToString(mixture),
+            probability
+          )
+        ).mkString(",")
+
 
       val evidence = samplesEvidence.allEvidences(input.index)
+
       val genotypeBuilder = new GenotypeBuilder()
+
       genotypeBuilder.name(input.sampleName)
+
       (input.tissueType, input.analyte) match {
-        case (TissueType.Normal, Analyte.DNA) => {
-          val germlineEvidence = samplesEvidence.allEvidences(input.index).asInstanceOf[NormalDNASingleSampleSingleAlleleEvidence]
+        case (TissueType.Normal, Analyte.DNA) =>
+          val germlineEvidence =
+            samplesEvidence
+              .allEvidences(input.index)
+              .asInstanceOf[NormalDNASingleSampleSingleAlleleEvidence]
+
           val posteriors = samplesEvidence.perNormalSampleGermlinePosteriors(input.index)
+
           val alleleGenotype = posteriors.maxBy(_._2)._1
-          genotypeBuilder.alleles(JavaConversions.seqAsJavaList(
-            Seq(makeHtsjdkAllele(alleleGenotype._1), makeHtsjdkAllele(alleleGenotype._2))))
-          genotypeBuilder.attribute("RL",
-            posteriors.map(pair => "[%s/%s]=%.8g".format(pair._1._1, pair._1._2, pair._2)).mkString(","))
+
+          genotypeBuilder.alleles(
```

Comment:
Echoing some of my motivation from a comment from #496:

> I like only having one thing happening on a given line; scala is so dense, and lines are so much wider than they are tall (in terms of pixels on screen), that things tend to be clearer the more lines they are split over, at least starting from where we are today.

The previous version of this line opens a bunch of `(` scopes and then closes 4 of them at once. One can do so much in Scala while passing params to a method (and indeed you have in this file and even on this line), each parameter-slot potentially being a whole code block that is arbitrarily complex, so it is really helpful to be explicit about which function-call's scope is opened and closed where in nested calls like this, is my feeling.

Parends in Scala end up grouping code in a more substantive way than we are used to, closer to what curly-braces do (in Scala and many other languages), which is why I like treating them more like we treat curly-braces; especially in code where we are not accumulating a lot of state in each expression (which is generally good!), curly braces often end up being superfluous while the parends actually carry all the weight of structuring the code.

For example, there are about 5 things I care about, as a reader, happening on this line:

```
Seq(makeHtsjdkAllele(alleleGenotype._1), makeHtsjdkAllele(alleleGenotype._2))
```

Start a `Seq`, make an allele, pass it the first element of the pair, make a second allele, pass it the second element, close the seq.

…all on a line that is about 30x as wide as it is tall, on my screen; that's just a bad way to lay out the information, imho.

> I often find scala's for harder to read than alternative ways of writing the same thing, so if there's a way to do this while keeping it a map that would be my preference

In cases with one vanilla `.map` and no destructuring, I agree with you, but any time you nest more than one map/flatMap:

```scala
foos.flatMap(foo => {
  bars.map(bar => {
    baz(bar)
  })
})
```

you incur much more cruft than:

```scala
for {
  foo <- foos
  bar <- bars
} yield
  baz(bar)
```

Similarly, when there is destructuring involved, as you outlined that you agree with about the case below, `for`s are a big win imho.

So that's basically where I come down: multiple nested `map`/`flatMap`s and/or ones involving destructuring are usually better as a `for`. That doesn't seem that far from what you seem to be saying.